pkg/kwokctl/runtime/util.go

```diff
@@ -116,10 +117,24 @@ func applyComponentPatch(component *internalversion.Component, patch internalver
 
 	component.Volumes = append(component.Volumes, patch.ExtraVolumes...)
 	component.Envs = append(component.Envs, patch.ExtraEnvs...)
-
 	for _, a := range patch.ExtraArgs {
-		component.Args = append(component.Args, fmt.Sprintf("--%s=%s", a.Key, a.Value))
+		if a.Override {
+			component.Args = applyComponentArgsOverride(component.Args, a)
+		} else {
```

Comment:
TODO: If there is no match, give a warning or return an error message. I will add a corresponding validation for all later.

Reply:
i'm adding the warning log, i think it's engough.